protos/client.proto

```diff
@@ -30,16 +30,32 @@ message Leaf {
     bytes data = 2;
 }
 
-// Possible commit statuses for a batch submitted to a validator
-//   * COMMITTED - the batch was accepted and has been committed to the chain
-//   * INVALID - the batch failed validation, it should be resubmitted
-//   * PENDING - the batch is still being processed
-//   * UNKNOWN - no status for the batch could be found (possibly invalid)
-enum BatchStatus {
-    COMMITTED = 0;
-    INVALID = 1;
-    PENDING = 2;
-    UNKNOWN = 3;
+// Information about the the status of a batch submitted to the validator.
+//
+// Attributes:
+//     batch_id: The id (header_signature) of the batch
+//     status: The committed status of the batch
+//     invalid_transaction: The header_signature of the txn that was rejected
+//     message: An message with additional information for an invalid batch
+//     extended_data: Any additional data included with an invalid batch
+//
+// Statuses:
+//     COMMITTED - the batch was accepted and has been committed to the chain
+//     INVALID - the batch failed validation, it should be resubmitted
+//     PENDING - the batch is still being processed
+//     UNKNOWN - no status for the batch could be found (possibly invalid)
+message BatchStatus {
+    enum Status {
+        COMMITTED = 0;
+        INVALID = 1;
+        PENDING = 2;
+        UNKNOWN = 3;
+    }
+    string batch_id = 1;
+    Status status = 2;
+    string invalid_transaction = 3;
```

Comment:
This possibly should be a list of invalid transactions, but I can't remember if batch processing fails fast.